x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -107,17 +107,13 @@ impl<'a, K: Key> State<'a, K> {
 
             let args_bytes = borsh::to_vec(&args).map_err(|_| StateError::Serialization)?;
 
-            let bytes = {
-                let ptr = unsafe { get_bytes(args_bytes.as_ptr(), args_bytes.len()) };
+            let ptr = unsafe { get_bytes(args_bytes.as_ptr(), args_bytes.len()) };
 
-                if ptr.is_null() {
-                    return Ok(None);
-                }
+            if ptr.is_null() {
+                return Ok(None);
+            }
```

Comment:
This is a leftover of https://github.com/ava-labs/hypersdk/pull/936, but it's probably not okay to return `Ok(None)` on any `err` from the host call to `get` (because we make the assumption that when this happens, the returned pointer is null). The reason is that `ErrNotFound` is not the only error that can be returned along the null pointer, there could also be `ErrClosed` and we should probably propagate that error rather than making the assumption that the value is not in the state.

Reply:
Posting here for posterity.

As discussed offline, we can't handle any transient errors from the host, as that would be a source of indeterminism. We use the null-pointer to signify no value and cast to `None`. Previously, we were serializing the `None` value as `[0]` and the `Some(<value>)` as `[1, <value-bytes>]`, but it's more efficient to skip serialization here. 

@iFrostizz, can you open an issue if we're returning `nil` for any case other than the value not being in the database?